spring-integration-core/src/main/java/org/springframework/integration/dsl/MessageProducerSpec.java

```diff
@@ -47,7 +50,9 @@ public MessageProducerSpec(@Nullable P producer) {
 	 */
 	@Override
 	public S id(@Nullable String id) {
-		this.target.setBeanName(id);
+		if (id != null) {
+			this.target.setBeanName(id);
```

Comment:
Just curious, why are we not throwing an error when they tryin to set the id to null, as it pertains to the bean name?

Reply:
That makes sense.   So a bean name does not have to match the id in that case.

Reply:
Could there be a case where:

```
mySpec.id("myID");
//beanName is now "myId"
//id is now "myId"
mySpec.id(null); //user calls id again and sets it to null
// beanName is "myId"
// id is null

Is this an issue?

Reply:
True. This is a problem.

Let me play with that if you are fine having more files to review in this already big PR 😄 
For example:
```
	public static AmqpMessageChannelSpec<?, ?> channel(ConnectionFactory connectionFactory) {
		return channel(null, connectionFactory);
	}

	public static AmqpMessageChannelSpec<?, ?> channel(@Nullable String id, ConnectionFactory connectionFactory) {
		return new AmqpMessageChannelSpec<>(connectionFactory)
				.id(id);
	}
```

So, the first method would not call the second just to have a nice delegation and code flow.
This way an `id` in the second method would not be null.

Reply:
Turned out that wasn't too bad.
Just only much more classes to review yet 😄

Reply:
Thank you! For reviewing this, spotting concern and raising your doubts.
I’m glad that I decided to not save you from boring code pushing the original change directly upstream 🤗